.github/workflows/development-environment.yml

```diff
@@ -37,43 +36,74 @@ jobs:
 
       # Trick for unattended Docker installations
       # https://github.com/docker/for-mac/issues/2359#issuecomment-943131345
-      - name: Install prerequisites
+      # NOTE: This can sometimes take up to 10 minutes
+      - name: Install Docker
         run: |
-          HOMEBREW_NO_AUTO_UPDATE=1 brew install --cask docker
+          HOMEBREW_NO_AUTO_UPDATE=1 brew install -v --cask docker
           sudo /Applications/Docker.app/Contents/MacOS/Docker --unattended --install-privileged-components
           open -a /Applications/Docker.app --args --unattended --accept-license
-          make prerequisites
 
-      - name: Setup Python
-        uses: ./.github/actions/setup-python
+      - name: Install missing brew packages
+        run: |
+          HOMEBREW_NO_AUTO_UPDATE=1 brew install -v libxmlsec1
 
-      - name: Cache (pyenv)
-        uses: actions/cache@v2
-        with:
-          path: ~/.pyenv
-          key: devenv-${{ matrix.os }}-pyenv-${{ hashFiles('.python-version') }}
+      - name: Install volta
+        run: |
+          curl https://get.volta.sh | bash
+
+      # This handles Python's cache
+      - name: Setup Python & cache
+        uses: ./.github/actions/setup-python
 
       - name: Cache (yarn)
         uses: actions/cache@v1 # We are explicitly using v1 due to perf reasons
         with:
           path: ${{ steps.info.outputs.yarn-cache-dir }}
-          key: devenv-${{ matrix.os }}-v1-yarn-${{ hashFiles('yarn.lock') }}
+          key: devenv-${{ runner.os }}-v1-yarn-${{ hashFiles('yarn.lock') }}
 
-      - name: Set up development environment (mostly as per docs)
+      # This tests starting up the dev services, loading mocks and pre-commit installation
+      # This can take over 15 minutes
+      - name: make bootstrap
         run: |
-          curl https://get.volta.sh | bash
           export VOLTA_HOME="$HOME/.volta"
           export PATH="$HOME/.volta/bin:$PATH"
+          [[ $(python -V) == "Python $(cat .python-version)" ]]
+          python -m venv .venv
+          source .venv/bin/activate
+          make bootstrap
+
+  # The pyenv set up takes long, thus, separating it into its own
+  pyenv-setup:
+    name: pyenv set up
+    runs-on: macos-11
+    timeout-minutes: 15
+    env:
+      # Make the shell to be Mac's default
+      SHELL: /bin/zsh
```

Comment:
Does this actually change the shell, or just the env var? Do we need something like https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_iddefaultsrun ?

Reply:
This is so this piece of code works:
https://github.com/getsentry/sentry/blob/47b837a69c38e190a7555de81e6d7d216498b854/scripts/pyenv_setup.sh#L17-L40

Reply:
I've updated the comment.